centos/centos-8.x/centos-8.3-hpc/install_mellanoxofed.sh

```diff
@@ -1,15 +1,39 @@
 #!/bin/bash
 set -ex
 
-MLNX_OFED_DOWNLOAD_URL=https://azhpcstor.blob.core.windows.net/azhpc-images-store/MLNX_OFED_LINUX-5.2-1.0.4.0-rhel8.3-x86_64.tgz
-TARBALL=$(basename ${MLNX_OFED_DOWNLOAD_URL})
+VERSION="5.8-1.0.1.1"
+$COMMON_DIR/write_component_version.sh "MOFED" $VERSION
+TARBALL="MLNX_OFED_LINUX-$VERSION-rhel8.3-x86_64.tgz"
+MLNX_OFED_DOWNLOAD_URL=https://content.mellanox.com/ofed/MLNX_OFED-${VERSION}/$TARBALL
 MOFED_FOLDER=$(basename ${MLNX_OFED_DOWNLOAD_URL} .tgz)
 
-$COMMON_DIR/download_and_verify.sh $MLNX_OFED_DOWNLOAD_URL "dd13ffa54524af9509e3de7288a863aa48ed7016bb3d15446523703020c41715"
+$COMMON_DIR/download_and_verify.sh $MLNX_OFED_DOWNLOAD_URL "370f49fd9190f046e1a41853e9f03508b57f1a29fa01f73099e564e6529a979d"
 tar zxvf ${TARBALL}
 
 KERNEL=( $(rpm -q kernel | sed 's/kernel\-//g') )
-KERNEL=${KERNEL[-1]}
-yum install -y kernel-devel-${KERNEL}
-./${MOFED_FOLDER}/mlnxofedinstall --kernel $KERNEL --kernel-sources /usr/src/kernels/${KERNEL} --add-kernel-support --skip-repo
+# Uncomment the lines below if you are running this on a VM
+# RELEASE=( $(cat /etc/centos-release | awk '{print $4}') )
+# yum install -y https://vault.centos.org/${RELEASE}/BaseOS/x86_64/os/Packages/kernel-devel-${KERNEL}.rpm
+# yum install -y https://vault.centos.org/${RELEASE}/BaseOS/x86_64/os/Packages/kernel-modules-extra-${KERNEL}.rpm
```

Comment:
hmm, what does this mean? 
Do we need to update the package source when building the image?

Reply:
Yes, kernel-devel and kernel-modules-extra are dependent on kernel version and so are our HPC components. The repo may have kernel packages which could result in unwanted updates (and sometimes may not have the required kernel packages). To avoid this, we explicitly download the kernel packages from official repos/ mirrors